test/federated/test_plan.py

```diff
@@ -15,30 +15,87 @@
 from syft.workers import WebsocketServerWorker
 
 
-def test_plan_built_locally(hook):
+def test_plan_built_automatically(hook):
     # To run a plan locally the local worker can't be a client worker,
     # since it needs to register objects
     hook.local_worker.is_client_worker = False
 
-    @sy.func2plan
+    @sy.func2plan(args_shape=[(1,)])
```

Comment:
Could we add support for variably shaped args? ie a tensor with one set dimension and the other unset (for things like batching)

Reply:
Yes, I've added support for -1, which usually means "unknown" or "any shape" tensors. Since for building the function we need to define the shape of the mock data, I map -1 to 1 in the mock data.